compiler/src/dotty/tools/dotc/typer/VarianceChecker.scala

```diff
@@ -132,14 +132,16 @@ class VarianceChecker()(implicit ctx: Context) {
           ctx.debuglog(s"Skipping variance check of ${sym.showDcl}")
         case tree: TypeDef =>
           checkVariance(sym, tree.pos)
+          tree.rhs match {
+            case rhs: Template => traverseChildren(rhs)
```

Comment:
Here `traverseChildren` sees `parents` (in the form of a constructor call), but this `Traverser` ignores terms and only looks at definitions...